internal/powerplatform/services/data_record/api_data_record.go

```diff
@@ -151,29 +165,93 @@ func (client *DataRecordClient) ApplyDataRecords(ctx context.Context, environmen
 
 	result.Id = locationHeader
 
-	/*
-		for key, value := range relations {
-			if nestedMapList, ok := value.([]interface{}); ok {
-				apiUrl := &url.URL{
+	for key, value := range relations {
+		if nestedMapList, ok := value.([]interface{}); ok {
+			apiUrl := &url.URL{
+				Scheme: "https",
+				Host:   strings.TrimPrefix(environmentUrl, "https://"),
+				Path:   fmt.Sprintf("/api/data/v9.2/%s(%s)/%s", entityDefinition.LogicalCollectionName, result.Id, key),
```

Comment:
Consider using a constant for the `DATAVERSE_API_VERSION` instead of hardcoding v9.2 everywhere

```suggestion
				Path:  path.Join("/api/data", DATAVERSE_API_VERSION, fmt.Sprintf("%s(%s)", entityDefinition.LogicalCollectionName, result.Id), key)
```

Reply:
done!